presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java

```diff
@@ -217,6 +217,49 @@ public void testGroupByInvalidOrdinal()
     }
 
     @Test
+    public void testGroupByWithSubquerySelectExpression()
+            throws Exception
+    {
+        analyze("SELECT (SELECT t1.a) FROM t1 GROUP BY a");
+        analyze("SELECT (SELECT a) FROM t1 GROUP BY t1.a");
```

Comment:
`SELECT a*2 FROM t1 GROUP BY t1.a*2` I added a test using _different_ expressions. I don't see value in testing _that_ we don't support something.